src/queues.c

```diff
@@ -121,6 +123,36 @@ static bool queues_notification_is_ready(const struct notification *n, struct du
                 return true;
 }
 
+/**
+ * Check if a notification has timed out
+ *
+ * @param n the notification to check
+ * @param status the current status of dunst
+ * @returns true, if the notification isn't timed out, otherwise false
+ */
+static bool queues_notification_is_finished(struct notification *n, struct dunst_status status)
+{
+        assert(n);
+
+        if (n->timeout == 0) // sticky
+                return true;
+        if (n->start == 0)   // hidden // TODO: is this really the implication of hidden?
+                return true;
+
+        /* don't timeout when user is idle */
+        if (status.idle && !status.fullscreen && !n->transient) {
+                n->start = time_monotonic_now();
```

Comment:
Why are you modifying `n->start` in a query function? What's the point of this if block entirely?

Reply:
You're never considered idle, if you're in fullscreen. Current master:

https://github.com/dunst-project/dunst/blob/ed341dfca30ed496bb056a00251af6a17d1f7a07/src/queues.c#L346

I think, I should use exactly this line in there also in the new function.